cmd/swctl/app/cmd_config.go

```diff
@@ -20,29 +21,56 @@ type ConfigCmdOptions struct {
 
 func (opts *ConfigCmdOptions) InstallFlags(flagset *pflag.FlagSet) {
 	flagset.BoolVar(&opts.ShowInternal, "show-internal", false, "Add Stonework internal configuration to output if possible")
+
 }
 
 func NewConfigCmd(cli Cli) *cobra.Command {
 	var (
 		opts ConfigCmdOptions
 	)
 	cmd := &cobra.Command{
-		Use:   "config [flags] ACTION",
-		Short: "Manage config of StoneWork components",
-		Args:  cobra.ArbitraryArgs,
-		// DisableFlagParsing: true,
+		Use:                "config [flags] ACTION",
+		Short:              "Manage config of StoneWork components",
+		Args:               cobra.ArbitraryArgs,
+		DisableFlagParsing: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
 			opts.Args = args
-			return runConfigCmd(cli, opts)
+			return runConfigCmd(cli, opts, cmd)
 		},
 	}
 	opts.InstallFlags(cmd.PersistentFlags())
 	return cmd
 }
 
-func runConfigCmd(cli Cli, opts ConfigCmdOptions) error {
+type temporaryFlag struct {
+	// full name flag
+	key string
+	// full line
+	value string
+}
+
+func runConfigCmd(cli Cli, opts ConfigCmdOptions, cmd *cobra.Command) error {
+	var ParsedFlags []temporaryFlag
 	args := opts.Args
 
+	if slices.Contains(args, "--help") || slices.Contains(args, "-h") {
+		// add Local flags to stdout of agentctl
+		stdout, stderr, shouldReturn, returnValue := mergeHelpers(cmd, ParsedFlags, cli, args)
+		if shouldReturn {
+			return returnValue
+		}
+
+		fmt.Fprintln(cli.Err(), stderr)
+		fmt.Fprintln(cli.Out(), stdout)
+		return nil
+	}
+
+	if slices.Contains(args, "--show-internal") {
+		opts.ShowInternal = true
+		idx := slices.Index[string](args, "--show-internal")
+		args = append(args[:idx], args[idx+1:]...)
+	}
```

Comment:
Why is this code block needed? Isn't `--show-internal` always passed to the command inside the `ShowInternal` field of the `ConfigCmdOptions` struct?

Reply:
line 35 is disabling flagParsing and these flags are taken as arguments. Only option I think is to process these flags like that. From my point of view its better because you can decide which flags belongs to swctl and which to agenctl .